File: 30-Day LeetCoding Challenge/Python/Day 4 - Move Zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        count = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                temp = nums[count]
                nums[count] = nums[i]
                nums[i] = temp
                count += 1
        
s = Solution()
nums = [0,1,0,3,12]
s.moveZeroes(nums)
print(nums)

File: 30-Day LeetCoding Challenge/C++/Day 4 - Move Zeroes.cpp
#include <iostream>
#include <vector>
#include "limits.h"
using namespace std;

class Solution {
    public:
        void moveZeroes(vector<int>& nums) {
            int count = 0;
            for(int i = 0; i < nums.size(); i++) {
                if(nums[i] != 0) {
                    //swap nums[i] and nums[count]
                    int temp = nums[count];
                    nums[count] = nums[i];
                    nums[i] = temp;
                    count++;
                }
            }
        }
};
int main() {
    Solution s;
    vector<int> nums{0,1,0,3,12};
    s.moveZeroes(nums);
    for(int num : nums) {
        cout << num << ", ";
    }

}